mir_eval/separation.py

```diff
@@ -324,11 +324,19 @@ def bss_eval_sources_framewise(reference_sources, estimated_sources,
     # k iterates across all the windows
     for k in range(nwin):
         win_slice = slice(k * hop, k * hop + window)
-        sdr[:, k], sir[:, k], sar[:, k], perm[:, k] = bss_eval_sources(
-            reference_sources[:, win_slice],
-            estimated_sources[:, win_slice],
-            compute_permutation
-        )
+        ref_slice = reference_sources[:, win_slice]
+        est_slice = estimated_sources[:, win_slice]
+        # check for a silent frame
+        if (not _any_source_silent(ref_slice) and
+                not _any_source_silent(est_slice)):
+            sdr[:, k], sir[:, k], sar[:, k], perm[:, k] = bss_eval_sources(
+                ref_slice,
```

Comment:
One last style nitpick for good measure - here and below, you can just put all the arguments on a single line.